### PR TITLE
Add test on repository delete

### PR DESCRIPTION
**Pull request by ismail1432**
Should fix https://github.com/mtarld/api-platform-ms-bundle/issues/63

### CODE REVIEW

File: tests/HttpRepository/HttpRepositoryTest.php [hunk 1]
```diff
@@ -469,4 +472,28 @@ public function testDeleteResourceWithoutIri(): void
...
+     */
+    public function testItThrowAnAppropriateErrorIfResponseItNotSuccess(int $statusCode, string $exceptionClassName)
+    {
+        $this->expectException($exceptionClassName);
+
+        $httpClient = new MockHttpClient([
+            new MockResponse('', ['http_code' => $statusCode]),
+        ]);
+
+        static::getContainer()->set('test.http_client', $httpClient);
```
Comment (mtarld) [inline tests/HttpRepository/HttpRepositoryTest.php:487]:
```suggestion
        static::$container->set('test.http_client', $httpClient);
```

File: tests/HttpRepository/HttpRepositoryTest.php [hunk 2]
```diff
@@ -469,4 +472,28 @@ public function testDeleteResourceWithoutIri(): void
...
+        /** @var PuppyHttpRepository $httpRepository */
+        $httpRepository = static::getContainer()->get(PuppyHttpRepository::class);
```
Comment (mtarld) [inline tests/HttpRepository/HttpRepositoryTest.php:489]:
```suggestion
        $httpRepository = static::$container->get(PuppyHttpRepository::class);
```

**Review by mtarld — COMMENTED** [review]
Really nice :slightly_smiling_face:

File: tests/HttpRepository/HttpRepositoryTest.php (hunk 2 above)
Comment (mtarld) [reply tests/HttpRepository/HttpRepositoryTest.php:489]: (And we'll handle the deprecation in #65 )

Comment (ismail1432) [reply tests/HttpRepository/HttpRepositoryTest.php:489]: And what about keep like this and merge #65 before this one?

File: tests/HttpRepository/HttpRepositoryTest.php [hunk 3]
```diff
@@ -469,4 +472,28 @@ public function testDeleteResourceWithoutIri(): void
         $httpRepository = static::$container->get(PuppyHttpRepository::class);
         $httpRepository->delete(new PuppyResourceDto(null, 'foo'));
     }
+
+    /**
+     * @dataProvider statusCodeAndExceptionProvider
+     */
+    public function testItThrowAnAppropriateErrorIfResponseItNotSuccess(int $statusCode, string $exceptionClassName)
```
Comment (ismail1432) [inline tests/HttpRepository/HttpRepositoryTest.php:479]:
```suggestion
    public function testItThrowAnAppropriateErrorIfResponseIsNotSuccess(int $statusCode, string $exceptionClassName)
```

**Review by mtarld — APPROVED** [review]

**Comment by mtarld** [comment]
Superseeded by https://github.com/mtarld/api-platform-ms-bundle/pull/69

Thank you very much @ismail1432 :slightly_smiling_face:

**Pull request closed without merging**